### PR TITLE
fix: added custom breakdown of one day to handle billing graph issue.

### PR DESCRIPTION
**Pull request by rahulkeswani101**
#### Related Issues / PR's

[1287](https://github.com/SigNoz/engineering-pod/issues/1287)

#### Screenshots

Before - 

image

After - 

image

### CODE REVIEW

File: frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx [hunk 1]
```diff
@@ -46,6 +46,19 @@ const calculateStartEndTime = (
 ): { startTime: number; endTime: number } => {
 	const timestamps: number[] = [];
 	data?.details?.breakdown?.forEach((breakdown: any) => {
+		console.log(breakdown);
```
Comment (SagarRajput-7) [inline frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx:49]: remove console

**Review by SagarRajput-7 — DISMISSED** [review]

**Comment by SagarRajput-7** [comment]
Please add images also for better understanding

File: frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx [hunk 2]
```diff
@@ -46,6 +46,18 @@ const calculateStartEndTime = (
 ): { startTime: number; endTime: number } => {
 	const timestamps: number[] = [];
 	data?.details?.breakdown?.forEach((breakdown: any) => {
+		if (breakdown?.dayWiseBreakdown?.breakdown?.length === 1) {
```
Comment (vikrantgupta25) [inline frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx:49]: add inline comment for the reasoning why are we doing this. it won't be trivial for someone to understand looking at this later.

**Review by vikrantgupta25 — APPROVED** [review]
LGTM

**Review by SagarRajput-7 — APPROVED** [review]

**Comment by srikanthccv** [comment]
@rahulkeswani101 can you pass dummy data and share what does it look like if the data exists for 2 days and 3days?

File: frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx [hunk 3]
```diff
@@ -58,6 +58,21 @@ const calculateStartEndTime = (
 
 export function BillingUsageGraph(props: BillingUsageGraphProps): JSX.Element {
 	const { data, billAmount } = props;
+	// Added this to fix the issue where breakdown with one day data are causing the bars to spread across multiple days
+	data?.details?.breakdown?.forEach((breakdown: any) => {
+		if (breakdown?.dayWiseBreakdown?.breakdown?.length === 1) {
+			const currentDay = breakdown.dayWiseBreakdown.breakdown[0];
+			const nextDay = {
+				...currentDay,
+				timestamp: currentDay.timestamp + 86400,
```
Comment (YounixM) [inline frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx:67]: Please add a comment on logic here

Comment (rahulkeswani101) [reply frontend/src/container/BillingContainer/BillingUsageGraph/BillingUsageGraph.tsx:67]: I have already added the comment for logic.

**Comment by rahulkeswani101** [comment]
> @rahulkeswani101 can you pass dummy data and share what does it look like if the data exists for 2 days and 3days?

@srikanthccv I am adding the screenshots for you to look at. Please let me know if we are good with this and can merge the PR.

image

image

**Review by srikanthccv — APPROVED** [review]
LGTM

**Review by YounixM — APPROVED** [review]

**Pull request merged**